common/nodes/database-schema.xml

```diff
@@ -405,6 +405,50 @@ CREATE TABLE ib_memberships (
   INDEX (switch, part_name, interface)
 );
 
+<!-- Firmware -->
+<!-- need to drop in reverse order of creation because of FK constraints -->
+DROP TABLE IF EXISTS firmware;
+DROP TABLE IF EXISTS firmware_model;
+DROP TABLE IF EXISTS firmware_make;
+DROP TABLE IF EXISTS firmware_imp;
+
+CREATE TABLE firmware_imp (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	name VARCHAR(255) NOT NULL,
+	INDEX (name),
+	CONSTRAINT unique_name UNIQUE (name)
+);
+
+CREATE TABLE firmware_make (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	name VARCHAR(255) NOT NULL,
+	INDEX (name),
+	CONSTRAINT unique_name UNIQUE (name)
+);
+
+CREATE TABLE firmware_model (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	name VARCHAR(255) NOT NULL,
+	make_id INT NOT NULL,
+	FOREIGN KEY (make_id) REFERENCES firmware_make(id) ON DELETE RESTRICT,
```

Comment:
You shouldn't need the `ON DELETE RESTRICT` here, since I am pretty sure that is the default action. But double check.